packages/plugins/plugin-inbox/src/components/Message/MessageContainer.tsx

```diff
@@ -2,23 +2,78 @@
 // Copyright 2025 DXOS.org
 //
 
-import React from 'react';
+import React, { useMemo } from 'react';
 
+import { DXN } from '@dxos/keys';
+import { resolveRef, useClient } from '@dxos/react-client';
+import { type Space } from '@dxos/react-client/echo';
+import { useThemeContext, type ThemedClassName } from '@dxos/react-ui';
+import {
+  createBasicExtensions,
+  createMarkdownExtensions,
+  createThemeExtensions,
+  decorateMarkdown,
+  preview,
+  useTextEditor,
+} from '@dxos/react-ui-editor';
 import { StackItem } from '@dxos/react-ui-stack';
+import { mx } from '@dxos/react-ui-theme';
 import { type MessageType } from '@dxos/schema';
 
+export type MessageProps = ThemedClassName<{
+  space?: Space;
+  message: MessageType;
+}>;
+
+const Message = ({ space, message, classNames }: MessageProps) => {
+  const client = useClient();
+  const { themeMode } = useThemeContext();
+
+  const content = useMemo(() => {
+    return message.blocks
+      .filter((block) => 'text' in block)
+      .map((block) => block.text)
+      .join('\n');
+  }, [message.blocks]);
+
+  // TODO(ZaymonFC): How to prevent caret and selection?
+  const extensions = useMemo(() => {
+    if (space) {
+      return [
+        createBasicExtensions({ readOnly: true, lineWrapping: true }),
+        createMarkdownExtensions({ themeMode }),
+        createThemeExtensions({ themeMode }),
+        decorateMarkdown(),
+        preview({
+          onLookup: async ({ label, ref }) => {
+            const dxn = DXN.parse(ref);
+            if (!dxn) {
+              return null;
+            }
+
+            const object = await resolveRef(client, dxn, space);
+            return { label, object };
+          },
```

Comment:
from what I could tell this is no longer needed, I think this was an old way this extension worked, you should be able to remove this

Reply:
I was copying `Transcript`, how does the extension get the client and space otherwise?

Reply:
the preview plugin binds that, that callback is never actually called anymore as far as I could tell

Reply:
check with @thure / @richburdon, I wasn't sure if it was new or old code so I left it

Reply:
I'll leave it until we overhaul `TextEditorPreview.stories.tsx`.